test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/EmptyAttributeTagHelpers.DesignTime.cs

```diff
@@ -0,0 +1,62 @@
+namespace TestOutput
+{
+    using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+    using System;
+    using System.Threading.Tasks;
+
+    public class EmptyAttributeTagHelpers
+    {
+        private static object @__o;
+        private void @__RazorDesignTimeHelpers__()
+        {
+            #pragma warning disable 219
+            string __tagHelperDirectiveSyntaxHelper = null;
+            __tagHelperDirectiveSyntaxHelper = 
+#line 1 "EmptyAttributeTagHelpers.cshtml"
+              "something"
+
+#line default
+#line hidden
+            ;
+            #pragma warning restore 219
+        }
+        #line hidden
+        private InputTagHelper __InputTagHelper = null;
+        private InputTagHelper2 __InputTagHelper2 = null;
+        private PTagHelper __PTagHelper = null;
+        #line hidden
+        public EmptyAttributeTagHelpers()
+        {
+        }
+
+        #pragma warning disable 1998
+        public override async Task ExecuteAsync()
+        {
+            __InputTagHelper = CreateTagHelper<InputTagHelper>();
+            __InputTagHelper.Type = "";
+            __InputTagHelper2 = CreateTagHelper<InputTagHelper2>();
+            __InputTagHelper2.Type = __InputTagHelper.Type;
+#line 4 "EmptyAttributeTagHelpers.cshtml"
+__InputTagHelper2.Checked = ;
```

Comment:
this case is valid HTML.  should handle convert empty `bool` values to `true`.  (`checked`, `checked=''`, and `checked='checked'` all mean `true`; everything else is `false`) .

otherwise, we should warn tag helper authors that Razor isn't smart enough to handle HTML setting `bool` tag helper properties correctly.  suggest they bind them as `string`s and handle the meaning themselves.  that's what the MVC tag helpers currently do (and at least one get's it wrong).